src/classes/Collision/Factory.js

```diff
@@ -0,0 +1,52 @@
+Goblin.Collision.Factory = {
+    /**
+     * returns a contact if objects collide with each other and null otherwise.
+     *
+     * @callback getContact
+     * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectA
+     * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectB
+     * @param {boolean} doLightweightCollision - if true then only basic collision information will be returned (w/o normal, penetration depth, etc).
+     * @returns {Goblin.ContactDetails|null}
+     */
+
+    /**
+     * @type {Array<[object, object, getContact]>}
+     */
+    _collisionTable: null,
+
+    _populateCollisionTable: function() {
+        Goblin.Collision.Factory._collisionTable = [
+            [ Goblin.SphereShape, Goblin.SphereShape, Goblin.Collision.sphereSphere ],
+            [ Goblin.SphereShape, Goblin.BoxShape, Goblin.BoxSphere ],
+            [ Goblin.SphereShape, Goblin.CapsuleShape, Goblin.Collision.sphereCapsule ],
+            [ Goblin.SphereShape, Goblin.ConvexHullShape, Goblin.Collision.sphereConvexHull ],
+        ];
+    },
+
+    /**
+     * @param {object} shapeA
+     * @param {object} shapeA.shape
+     * @param {object} shapeB
+     * @param {object} shapeB.shape
+     * @returns {getContact}
+     */
+    getCollisionMethod: function( shapeA, shapeB ) {
```

Comment:
Why not hardcoding the array (after lazy initialization)? Also, the loop below seems excessive – I'd use a single array with an indexing rule like (shape1 << 8) + shape2

Reply:
Not sure what "hardcoding" means in this case. It's populated during the first invocation of `getCollisionMethod`. We cannot populate it's earlier because some collision methods (such as `sphereCapsule`) may not be added to `Goblin.Collision` object yet. Another possible solution is to explicitly include list all collision method below `Factory` in the `gulpfile.js`, but it's quite a burden IMO. This will be resolved if we decide to go with `webpack` / `google closure compiler`.

Reply:
> I'd use a single array with an indexing rule like (shape1 << 8) + shape2
Unfortunately, it's not a C++ code and there is no way to convert shapes to numbers

Reply:
fixed